tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -605,6 +605,8 @@ class CompilerTest {
   TEST_METHOD(CodeGenShare_Mem2)
   TEST_METHOD(CodeGenShare_Mem2Dim)
   TEST_METHOD(CodeGenShift)
+  TEST_METHOD(CodeGenShortCircuiting0)
+  TEST_METHOD(CodeGenShortCircuiting1)
   TEST_METHOD(CodeGenSimpleDS1)
```

Comment:
We should also add a test for example 3 from #332, to make sure we don't break it in the future.

```
// Example 3

bool inc(inout float4 v) { v++; return true; }

float4 main(int val: A, float4 color: COLOR) : SV_TARGET {
  if (inc(color) || inc(color)) return color;

  return color + 4.2;
}
```